core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -418,19 +201,12 @@ public long nowInMillis() {
         return System.currentTimeMillis();
     }
 
-    public NestedScope nestedScope() {
-        return nestedScope;
-    }
-
-    /**
-     * Return whether the setting is deprecated.
-     */
-    public boolean isDeprecatedSetting(String setting) {
-        return parseFieldMatcher.match(setting, CACHE) || parseFieldMatcher.match(setting, CACHE_KEY);
+    public MapperService mapperService() {
+        return indexQueryParser.mapperService;
     }
 
-    public Version indexVersionCreated() {
-        return indexVersionCreated;
+    //norelease remove dependency on context here
+    public MappedFieldType fieldMapper(String name) {
```

Comment:
since we have a getter for the context, can't we simply remove and replace callers code with queryParseContext.context().fieldMapper(name) ?